src/compiler/core.ts

```diff
@@ -1097,6 +1097,15 @@ namespace ts {
         return result;
     }
 
+    /**
+     * Creates a set from the elements of an array.
+     *
+     * @param array the array of input elements.
+     */
+    export function arrayToSet<T>(array: T[], makeKey: (value: T) => string): Map<true> {

```

Comment:
It's generally useful to factor generic code out of a more specific context. That way you don't have to worry about the specifics of how an array is converted to a set, and only have to worry about the actual problem domain.